lib/src/migrators/division.dart

```diff
@@ -244,40 +272,45 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
       return false;
     }
 
-    ListExpression? channels;
-    if (node.arguments.positional.length == 1 &&
-        node.arguments.named.isEmpty &&
-        node.arguments.positional.first is ListExpression) {
-      channels = node.arguments.positional.first as ListExpression?;
-    } else if (node.arguments.positional.isEmpty &&
-        node.arguments.named.containsKey(r'$channels') &&
-        node.arguments.named.length == 1 &&
-        node.arguments.named[r'$channels'] is ListExpression) {
-      channels = node.arguments.named[r'$channels'] as ListExpression?;
-    }
-    if (channels == null ||
-        channels.hasBrackets ||
-        channels.separator != ListSeparator.space ||
-        channels.contents.length != 3 ||
-        channels.contents.last is! BinaryOperationExpression) {
-      return false;
-    }
+    var channels = switch (node.arguments) {
+      ArgumentInvocation(positional: [ListExpression arg, ...], named: Map()) =>
+        arg,
+      ArgumentInvocation(
+        positional: [],
+        named: {r'$channels': ListExpression arg}
+      ) =>
+        arg,
+      _ => null,
+    };
 
-    var last = channels.contents.last as BinaryOperationExpression;
-    if (last.left is! NumberExpression || last.right is! NumberExpression) {
-      // Handles cases like `rgb(10 20 30/2 / 0.5)`, since converting `30/2` to
-      // `divide(30, 20)` would cause `/ 0.5` to be interpreted as division.
-      _patchSpacesToCommas(channels);
-      _patchOperatorToComma(last);
+    if (channels
+        case ListExpression(
+          hasBrackets: false,
+          separator: ListSeparator.space,
+          contents: [_, _, BinaryOperationExpression last]
+        )) {
+      // Handles cases like `rgb(10 20 30/2 / 0.5)`, since converting `30/2`
+      // to `div(30, 20)` would cause `/ 0.5` to be interpreted as division.
+      switch (last) {
+        case BinaryOperationExpression(
+            left: NumberExpression _,
+            right: NumberExpression _
```

Comment:
Here I think `NumberExpression()` is more idiomatic

Reply:
Fixed